### ISSUE TITLE
/plugins/BadChannelName.php - ERROR

### ISSUE THREAD

**Issue by HellBz** [JABirchall/NimdaTS3#28]
PHP Fatal error:  Call to a member function contains() on integer in /var/www/virtual/hellbz.de/htdocs/ts3bot/plugins/BadChannelName.php on line 23

**Comment by Najsr** [comment]
Temporarily fixed by my commit (https://github.com/JABirchall/NimdaTS3/commit/a150982257b5dcdb150c81967acf02d1e39de60b)

**Comment by JABirchall** [comment]
The correct fix would be to return the contructed StringHelper here: https://github.com/JABirchall/NimdaTS3/blob/master/vendor/fkubis/teamspeak-php-framework/TeamSpeak3/Node/Channel.php#L585

and that would allow for
```
if(!$this->info->contains($regex, true)) {
     continue;
}
```
However I dont like this since $info would be the channel instance and its not informative.

Something I always wanted to do which I feel is needed in the teamspeak framework is creating a repository for getting and setting node properties. That way each property can be corrected contructed to its type and can be documents for devs to know what that can and cant access.
I know when i started developing bots for teamspeak I was using alot of var_dump() to see what the classes contained which i could use.

**Comment by JABirchall** [comment]
In the mean time the temp fix is in 0.14.2 https://github.com/JABirchall/NimdaTS3/releases/tag/v0.14.2
